packages/docusaurus/src/client/ClientLifecyclesDispatcher.tsx

```diff
@@ -46,8 +46,12 @@ function ClientLifecyclesDispatcher({
         const element = document.getElementById(id);
         element?.scrollIntoView();
       }
-      dispatchLifecycleAction('onRouteDidUpdate', {previousLocation, location});
+      return dispatchLifecycleAction('onRouteDidUpdate', {
+        previousLocation,
+        location,
+      });
```

Comment:
Exposed the cleanup function here, so we can clean up the side-effect of appending DOM child.